[triggerType and waveConfig] implmented triggerType for waveConfig an…

Alfredoeb9 opened the pull request:
…d tests

implemented triggerType type and updated waveConfig to take in new type, updated test for triggerType

<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is filled out.
Your pull requests will be merged after one of the collaborators approves.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [x ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ⌨️ Accessibility improvement
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

close #56286

### 💡 Background and Solution

- Request was sent to add a triggerType to the waveConfig for press effects

### 📝 Change Log
- 

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 🆕 WaveConfig now includes types for press, mouse and click events         |
| 🇨🇳 Chinese | 🆕  WaveConfig 现在包括按、鼠标和单击事件的类型  |

Alfredoeb9 committed:
[triggerType and waveConfig] implmented triggerType for waveConfig and tests

implemented triggerType type and updated waveConfig to take in new type, updated test for triggerType

wanpan11 commented:
Perform local snapshot update

<img width="271" height="291" alt="image" src="https://github.com/user-attachments/assets/adec64c5-b92e-46cb-bc5e-e7fd8ebe76e7" />

Additionally, this PR seems to only contain type definitions.

thinkasany committed:
Merge branch 'feature' into feature-branch

Alfredoeb9 committed:
Merge branch 'feature' into feature-branch

elrrrrrrr committed:
fix: setup utoo (#56409)

Co-authored-by: binghui.dbh <binghui.dbh@antgroup.com>

meet-student committed:
chore: remove outdated parse5 dependencies from package.json (#56410)

dependabot[bot] committed:
chore(deps-dev): bump father from 4.6.12 to 4.6.13 (#56416)

Bumps [father](https://github.com/umijs/father) from 4.6.12 to 4.6.13.
- [Release notes](https://github.com/umijs/father/releases)
- [Commits](https://github.com/umijs/father/compare/v4.6.12...v4.6.13)

---
updated-dependencies:
- dependency-name: father
  dependency-version: 4.6.13
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

li-jia-nan committed:
site: replace isCancelled with cancelledRef (#56415)

wanpan11 committed:
test: update

wanpan11 committed:
Merge branch 'feature' into feature-branch

wanpan11 committed:
docs: Optimize the dark mode of the homepage. (#56418)

meet-student committed:
Revert "chore: remove outdated parse5 dependencies from package.json (#56410)" (#56422)

This reverts commit eac8dde1e50a273ca92ec7dad654fd30a7caf707.

Alfredoeb9 committed:
[triggerTypes on wave] implemented new triggerTypes for wave util
* added type definitions for pointers and mouse
* changed event for onClick function as a Event and showWave as a mouseEvent based on triggerTypes
* created object that listens to triggerTypes
* created unit test to for triggeTypes
